### ISSUE TITLE
PB blocks downloading files from google docs

### ISSUE THREAD

**Issue by mdunphy** [EFForg/privacybadgerchrome#534]
Hi,
PB blocks downloading files from google docs. Manually switching the slider from yellow to green only sometimes fixes it. For example, when trying to download a file fails, checking PB shows an entry like:
  doc-0g-9s-docs.googleuserco...
with the slider on yellow. If I push it to green, and try to redownload the file, it will work if google uses the same server to service the request, but if they use a different server PB gets another entry like
 doc-00-9s-docs.googleuserco...
with yellow slider. So there's too many hosts to whitelist by hand each time. 

PB version 2015.8.5.1

**Comment by johnalexhebert** [comment]
Have you tested this using regex for PB whitelist entries? Example: *.googleusercontent.com

**Comment by cooperq** [comment]
@mdunphy could you try going the settings page and whitelisting all of googleusercontent.com?

**Comment by mdunphy** [comment]
I tried adding *.googleusercontent.com and and then just googleusercontent.com to the whitelist but it still blocks downloads.

There are a bunch of entries in the "User Filter Tab" to the right of the whitelist that have [something].googleusercontent.com with a yellow blocking slider, which seem to be overriding the whitelist. Some of them had a "return control to PB" button, which are presumably ones that I manually set before, so I returned all of those to PB, but a number remain there and it still blocks downloads.

**Comment by cooperq** [comment]
Okay, so I think there is some confusion here, @johnalexhebert @mdunphy ,  The domain whitelist is a list of first party domains on which privacy badger should not run at all. Adding googleusercontent.com here does nothing because it is not the first party domain, in this case docs.google.com would be the first party domain. I think the actual workaround here is to move the slider for googleusercontent.com in the options tab from red or yellow to green. That should hopefully fix the problem.

**Comment by mdunphy** [comment]
I confirm that moving the switch for "googleusercontent.com" to green fixes the issue

**Comment by anand-bhat** [comment]
The only change I had to do to get this working was to move the switch for "doc-0g-24-docs.googleusercontent.com" from yellow to green. Looking at the subdomain name, it probably is one among several.

![image](https://cloud.githubusercontent.com/assets/7332442/10120758/7c07fae6-649a-11e5-8d60-186ef0a1f780.png)

**Comment by DaveLessnau** [comment]
I can't download from Google Drive and under PB's settings, googleusercontent.com is already slid to the green side.  Yet, in the PB panel on drive.google.com, all 5 of the googleusercontent.com sliders (doc-0c-84-docs, lh3, lh5, lh6, and oath) show as yellow.

**Comment by cooperq** [comment]
moving googleusercontent.com to green in privacy badgers settings should whitelist all googleusercontent.com subdomains which fixes this problem. We should probably add a rule to domainExceptions to ask users to whitelist googleusercontent.com if they want to download things from google drive.

**Comment by DaveLessnau** [comment]
Maybe it's a combination of things on my system.  But, again, "googleusercontent.com" is green under PB's settings.  Yet, the individual instantiations of it on the PB tracker panel while on the Google Drive web page are yellow.  If I turn off PB, the problem goes away.  If I slide the individual sliders for the yellow instantiations on the PB panel to green, I can download from Google Drive.  Once.  At that point, the list of trackers updates, and a new, yellow, instantiation of googleusercontent.com show up and I've got to do it all over again.

Privacy Badger 2015.8.14
Google Chrome Version 47.0.2526.49 beta (64-bit)
Linux Mint 17.2 Cinnamon (64-bit)

**Comment by DaveLessnau** [comment]
Another data point:  I booted back into Windows 10 and tried downloading documents from Google Drive from Google Chrome.  Same problem.  BUT, the slider in settings for googleusercontent.com was still yellow.  I switched it to green, all the individual googleusercontent.coms in the tracking panel switched to green, and I could download.  Unfortunately, the problem persists on the Linux side of things.
